alpine-nix-rails-nginx/example-rails-template.rb

```diff
@@ -26,16 +26,42 @@
 web: bundle exec puma --quiet --config config/puma.rb
 PROCFILE
 
+file 'config/postgres.sh', <<-'POSTGRES'
+#!/usr/bin/env bash
+
+set -o errexit -o pipefail -o noglob -o noclobber -o nounset
+IFS=$'\n\t'
+
+export PATH="/nix/var/nix/profiles/default/bin:/nix/var/nix/profiles/default/sbin:$PATH"
+
+exec chpst -u postgres -- postgres -D $PGDATA
+POSTGRES
+
 file 'Dockerfile', <<-'DOCKERFILE'
 FROM dkubb/alpine-nix-rails-nginx:latest
 MAINTAINER Dan Kubb <dkubb@fastmail.com>
 
 ENV RAILS_ENV development
+ENV PGDATA    /var/lib/postgresql/data
 
-RUN nix-env --install sqlite-3.8.11.1
+COPY config/postgres.sh /etc/sv/postgres
+RUN source /etc/profile.d/nix.sh \
+  && /root/setup-directories.sh root /etc/sv /etc/service/postgres \
+  && /root/setup-directories.sh postgres "$(dirname "$PGDATA")" "$PGDATA" \
+  && chmod u+x /etc/sv/postgres \
+  && ln -s -- /etc/sv/postgres /etc/service/postgres/run
 
+# Setup default user and database
+USER postgres
 RUN source /etc/profile.d/nix.sh \
```

Comment:
I have to spin up pg to add the users, and the reason I spin it down is so the data is persisted. In a Dockerfile you can't rely on an earlier `RUN` command to have started a process; you can only rely on changes persisted to disk.